x-pack/plugins/features/server/feature_registry.ts

```diff
@@ -5,34 +5,66 @@
  */
 
 import { cloneDeep, uniq } from 'lodash';
-import { FeatureConfig, Feature, FeatureKibanaPrivileges } from '../common';
-import { validateFeature } from './feature_schema';
+import {
+  FeatureConfig,
+  Feature,
+  FeatureKibanaPrivileges,
+  ElasticsearchFeatureConfig,
+  ElasticsearchFeature,
+} from '../common';
+import { validateKibanaFeature, validateElasticsearchFeature } from './feature_schema';
 
 export class FeatureRegistry {
   private locked = false;
-  private features: Record<string, FeatureConfig> = {};
+  private kibanaFeatures: Record<string, FeatureConfig> = {};
+  private esFeatures: Record<string, ElasticsearchFeatureConfig> = {};
 
-  public register(feature: FeatureConfig) {
+  public registerKibanaFeature(feature: FeatureConfig) {
     if (this.locked) {
       throw new Error(
         `Features are locked, can't register new features. Attempt to register ${feature.id} failed.`
       );
     }
 
-    validateFeature(feature);
+    validateKibanaFeature(feature);
 
-    if (feature.id in this.features) {
+    if (feature.id in this.kibanaFeatures || feature.id in this.esFeatures) {
```

Comment:
> I was thinking it may be possible to add a different prefix to kibana features and elasticsearch features to prevent this type of collision, but I don't know what the benefit is if consumers

This would be a risky change to make, since UI Capabilities do not have a whole lot of type safety. It'd be easy to overlook a rename of this magnitude

Reply:
Ah, makes sense.